marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/MappingStepDefinitionImpl.java

```diff
@@ -16,15 +16,21 @@
 
 package com.marklogic.hub.step.impl;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.marklogic.hub.HubConfig;
+import com.marklogic.hub.impl.Versions;
 import com.marklogic.hub.step.AbstractStepDefinition;
+import com.marklogic.hub.util.BeanUtil;
 
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 
 public class MappingStepDefinitionImpl extends AbstractStepDefinition {
 
+    @JsonIgnore
+    Versions versions = BeanUtil.getBean(Versions.class);
```

Comment:
Oof. This is rough, but I understand the need for it. I definitely recommend not declaring this as a class attribute - there's no real performance again in that, as getBean is just hitting an in-memory hash map. Then you don't need JsonIgnore.

Reply:
Fixed in latest commit 69488d8bda278aa2ab8b428bab6f587ad18f8b89